pygmt/tests/test_grdimage.py

```diff
@@ -139,10 +139,10 @@ def test_grdimage_grid_and_shading_with_xarray(grid, xrgrid):
     """
     fig_ref, fig_test = Figure(), Figure()
     fig_ref.grdimage(
-        grid="@earth_relief_01d_g", region="GL", cmap="geo", shading=xrgrid
+        grid="@earth_relief_01d_g", region="GL", cmap="geo", shading=xrgrid, cores=2
     )
```

Comment:
This line is raising an error at https://github.com/GenericMappingTools/pygmt/actions/runs/7404780202/job/20146784435?pr=2945#step:8:778:

```
 E           pygmt.exceptions.GMTCLibError: Module 'grdimage' failed with status code 72:
Error:      grdimage [ERROR]: Option -x given more than once
Error:      grdimage [ERROR]: Option -x parsing failure. Correct syntax:
Error:      grdimage [ERROR]: Offending option -x2
```

Equivalent GMT CLI command also fails:

```
gmt grdimage @earth_relief_01d_g -RGL -Cgeo -x2 -png map
```

```
grdimage [ERROR]: Option -x given more than once
grdimage [ERROR]: Option -x parsing failure. Correct syntax:

   [-x[[-]<n>]] .
     Limit the number of cores used in multi-threaded algorithms [Default uses all 16 cores]. If <n> is negative then we select (16 - <n>) cores (or at least 1).
grdimage [ERROR]: Offending option -x2
```

Might be an upstream GMT bug?

Reply:
very likely

Reply:
Ok, reported upstream at https://github.com/GenericMappingTools/gmt/issues/8256. I've reverted the `grdimage` commit at 08f36d2b3eaa5ba6181412afbc0af0b2dc56bdcd for now.